### ISSUE TITLE
Allows to mutate `const` with non-null assertion

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#21848]
**TypeScript Version:**  2.8.0-dev.20180209

**Code**

```ts
declare const x: number | undefined;
x!++;
```

**Expected behavior:**

```
src/a.ts(2,2): error TS2540: Cannot assign to 'x' because it is a constant or a read-only property.
```

**Actual behavior:**

No error.

**Comment by DanielRosenwasser** [comment]
But you said it was a constant?

**Comment by andy-ms** [comment]
The *expected* behavior is an error. The actual behavior is no error. For `x: number` and `x++` there is correctly an error.

**Comment by DanielRosenwasser** [comment]
Sorry. It's been a long week.

**Comment by mhegazy** [comment]
thanks @jack-williams!

**Comment by jack-williams** [comment]
No problem!
